lib/email/sender.rb

```diff
@@ -228,6 +229,12 @@ def send
 
       email_log.message_id = @message.message_id
 
+      # Log when a message is being sent from a group SMTP address, so we
+      # can debug deliverability issues.
+      if from_address && smtp_group = Group.find_by(email_username: from_address, smtp_enabled: true)
```

Comment:
No need to load the ActiveRecord object here. Just plucking the group id will be sufficient for our needs :)